R/art.con.internal.R

```diff
@@ -0,0 +1,320 @@
+  art.con.df
+}
+
+### removes variables from the model formula (m.f) that are in the contrast formula (f)
+### replaces them with the new concatenated (the concatenation of all variables in the contrast formula)
+### creates and returns art model on art.concatenated.df using the created formula
+### note: this is not the same as using the full factorial model of all columns in df
+###       there can be columns in df that are not used in the model.
+#' @importFrom stringi stri_join
+generate.art.concatenated.model = function(m.f, m.f.parsed, art.concatenated.df, f.parsed){
+  
+  # fixed variables in model formula
+  m.f.fixed.variables = m.f.parsed$fixed.variables
+  # interaction variables
+  f.interaction.variables = f.parsed$interaction.variables
+  # concatenated interaction variable
+  f.concat.interaction.variable = f.parsed$concat.interaction.variable
+  
+  # indices of interaction variables in model formula fixed variable list
+  # e.g. f.interaction.variables = (a, c) and m.f.fixed.variables = (a, b, c) -> c(1, 3)
+  interaction.variable.index = match(f.interaction.variables, m.f.fixed.variables)
+  # remove interaction variables from model formula
+  # e.g. f.interaction.variables = (a, c) and m.f.fixed.variables = (a, b, c) -> list(b)
+  m.f.fixed.variables.no.interaction.vars = m.f.fixed.variables[-interaction.variable.index]
+  m.f.fixed.variables.with.concat = c(m.f.fixed.variables.no.interaction.vars, f.concat.interaction.variable)
+  
+  # create formula with response m.f.response, fixed vars m.f.fixed.variables.with.concat
+  # grouping variables m.f.grouping.variables, error variables m.f.error.variables
+  m.f.response = m.f.parsed$response
+  m.f.grouping.variables = m.f.parsed$grouping.variables
+  m.f.error.variables = m.f.parsed$error.variables
+  
+  # collapse fixed variable vector into string separated by *
+  # e.g. list(ac,b) -> "ac*b"
+  m.f.fixed.str = paste(m.f.fixed.variables.with.concat, collapse = "*")
+  # add parentheses around each grouping variable and 
+  # collapse grouping variables vector into string separated by +
+  # e.g list(1|d, 1|e) -> "(1|d) + (1|e)"
+  # note: empty list coerced to empty vector i.e list() -> character(0)
+  m.f.grouping.paren = if(length(m.f.grouping.variables) == 0) character() else paste("(", m.f.grouping.variables, ")", sep="")
+  m.f.grouping.str = if(length(m.f.grouping.paren) == 0) character() else paste(m.f.grouping.paren, collapse = "+")
+  # collaprse error variables vector into string separated by +
+  # e.g. list(Error(g), Error(h)) -> "Error(g) + Error(h)"
+  # note: empty list coerced to empty vector i.e list() -> character(0)
+  m.f.error.str = if(length(m.f.error.variables) == 0) character() else paste(m.f.error.variables, collapse = "+")
+  # assemble concatenated art formula
+  art.concatenated.formula.rhs.str = stri_join(m.f.fixed.str, m.f.grouping.str, m.f.error.str, sep="+", ignore_null= TRUE)
```

Comment:
You should be able to use `paste` instead of `stri_join` here, then you can drop the stringi dependency